pyannote/audio/core/task.py

```diff
@@ -221,8 +290,17 @@ def __init__(
         self.protocol, checks = check_protocol(protocol)
         self.has_validation = checks["has_validation"]
         self.has_scope = checks["has_scope"]
+        if not self.has_scope:
+            raise ValueError(
+                "Protocol must provide 'scope' information (e.g. 'file', 'database', or 'global')."
+            )
+
         self.has_classes = checks["has_classes"]
 
+        # metadata cache
+        self.cache = Path(cache) if cache else Path(mktemp())
```

Comment:
Looks like `mktemp` is deprecated: https://docs.python.org/3/library/tempfile.html#deprecated-functions-and-variables

Can we use something else?

Reply:
> Can we use something else?

I think we can directly build a random string by using `random` module . This will have the same effect as `mktemp()`

Reply:
I am pretty sure Python standard library has a proper function to replace `mktemp`.
Maybe in `pathlib`?

Reply:
We can use `mkstemp()` from `tempfile`